python/dgl/graphbolt/impl/neighbor_sampler.py

```diff
@@ -244,10 +258,56 @@ def __init__(
         prob_name,
         deduplicate,
         sampler,
+        layer_dependency=None,
+        batch_dependency=None,
     ):
+        if sampler.__name__ == "sample_layer_neighbors":
```

Comment:
Instead of checking the sampler name here, why not move the self._init_seed(batch_dependency) to __init__ of LayerNeighborSampler?